### PR TITLE
Add VGGFace2 format support

### PR DESCRIPTION
**Pull request by yasakova-anastasia**
### Summary

Added support for VGGFace2 format

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [x] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [x] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: datumaro/plugins/vgg_face2_format.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+class VggFace2Path:
+    ANNOTATION_DIR = "bb_landmark"
+    IMAGE_EXT = '.jpg'
+    BBOXES_FILE = 'loose_bb_'
+    LANDMARKS_FILE = 'loose_landmark_'
+
+class VggFace2Extractor(SourceExtractor):
+    def __init__(self, path):
+        if not osp.isfile(path):
+            raise Exception("Can't read annotation file '%s'" % path)
```
Comment (zhiltsov-max) [inline datumaro/plugins/vgg_face2_format.py:25]: I suggest changing the message, or adding `Expected a '.csv' annotation file`.

File: datumaro/plugins/vgg_face2_format.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        self._path = path
+        self._dataset_dir = osp.dirname(osp.dirname(path))
+
+        subset = osp.splitext(osp.basename(path))[0]
+        match = re.fullmatch(r'loose_landmark_\S+', subset)
```
Comment (zhiltsov-max) [inline datumaro/plugins/vgg_face2_format.py:30]: I feel like we need to use the constant above here. How about `startswith()`?

File: datumaro/plugins/vgg_face2_format.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+        items = {}
+        with open(path) as content:
+            landmarks_table = list(csv.DictReader(content))
+
+        for row in landmarks_table:
+            item_id = row['NAME_ID']
+            image_path = osp.join(self._dataset_dir, self._subset, item_id
+                + VggFace2Path.IMAGE_EXT)
+            annotations = []
+            if len([p for p in row if row[p] == '']) == 0:
```
Comment (zhiltsov-max) [inline datumaro/plugins/vgg_face2_format.py:52]: There is a visibility flag for missing points. Also we need to check if the number of points is different from 5.

**Review by zhiltsov-max — CHANGES_REQUESTED** [review]

**Review by zhiltsov-max — APPROVED** [review]
As I understood from the dataset description, there are also [attributes](https://github.com/ox-vgg/vgg_face2/tree/master/attributes), face direction and age properties somewhere.

**Comment by yasakova-anastasia** [comment]
Yes, I think it is. But the official website of this dataset is not available and I don’t know how to get more information.

**Pull request merged**